### ISSUE TITLE
Error with tf.losses.computeWeightedLoss: Cannot compute gradient: gradient function not found for NotEqual. 

### ISSUE THREAD

**Issue by medric49** [tensorflow/tfjs#7710]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow.js): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- TensorFlow.js installed from (npm or script link): npm
- TensorFlow.js version (use command below): 4.2.0
- Browser version: Chrome 112.0.5615.49

**Describe the current behavior**
I am using `tf.losses.computeWeightedLoss` loss function in the .compile() method and I get the error message:
Cannot compute gradient: gradient function not found for NotEqual.
It seems it is not possible to compute the gradient of this loss function.

**Standalone code to reproduce the issue**

```
const tf = require('@tensorflow/tfjs')

const model = tf.sequential({
    layers: [
      tf.layers.dense({inputShape: [784], units: 32, activation: 'relu'}),
      tf.layers.dense({units: 10, activation: 'softmax'}),
    ]
   });

   model.weights.forEach(w => {
    const newVals = tf.randomNormal(w.shape);
    // w.val is an instance of tf.Variable
    w.val.assign(newVals);
  });

  model.compile({
    optimizer: 'sgd',
    loss: tf.losses.computeWeightedLoss,
    metrics: ['accuracy']
  });


  // Generate dummy data.
const data = tf.randomNormal([100, 784]);
const labels = tf.randomUniform([100, 10]);

function onBatchEnd(batch, logs) {
  console.log('Accuracy', logs.acc);
}

// Train for 5 epochs with batch size of 32.
model.fit(data, labels, {
   epochs: 5,
   batchSize: 32,
   callbacks: {onBatchEnd}
 }).then(info => {
   console.log('Final accuracy', info.history.acc);
 });
```

**Other info / logs**

```
/home/project/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:3329
            throw new Error("Cannot compute gradient: gradient function not found " +
                  ^

Error: Cannot compute gradient: gradient function not found for NotEqual.
```

**Comment by gaikwadrahul8** [comment]
Hi, @medric49 

Apologize for the delayed response and I was able to replicate the same issue with same error which you mentioned above so we will have to dig more into this issue so we will update you soon, thank you for noticing this issue and I really appreciate your efforts and time. for your reference I have added screenshot below. Thank you!

```
gaikwadrahul-macbookpro:TFJS gaikwadrahul$ node test.js

============================
Hi, looks like you are running TensorFlow.js in Node.js. To speed things up dramatically, install our node backend, visit https://github.com/tensorflow/tfjs-node for more details. 
============================
/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:3374
            throw new Error("Cannot compute gradient: gradient function not found " +
                  ^

Error: Cannot compute gradient: gradient function not found for NotEqual.
    at _loop_1 (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:3374:19)
    at backpropagateGradients (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:3411:9)
    at /Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:5161:13
    at /Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:4562:22
    at Engine.scopedRun (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:4572:23)
    at Engine.tidy (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:4561:21)
    at Engine.gradients (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:5157:21)
    at variableGrads (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:13670:21)
    at Optimizer.computeGradients (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:23813:16)
    at Optimizer.minimize (/Users/gaikwadrahul/Desktop/TFJS/node_modules/@tensorflow/tfjs/node_modules/@tensorflow/tfjs-core/dist/tf-core.node.js:23765:23)

Node.js v18.15.0
gaikwadrahul-macbookpro:TFJS gaikwadrahul$ 

```

**Comment by Varun-S10** [comment]
Hi @medric49, Thank you for your patience. We have reproduced the issue. As a workaround, could you please try using a standard loss function like 'categoricalCrossentropy' instead of tf.losses.computeWeightedLoss? Please let us know if this resolves the issue for you. 
```
model.compile({
  optimizer: 'sgd',
  loss: 'categoricalCrossentropy', // or 'meanSquaredError' depending on your task
  metrics: ['accuracy']
});
```
